editor/src/clj/editor/game_object_common.clj

```diff
@@ -224,8 +224,8 @@
     {:resource build-resource
      :content (protobuf/map->bytes GameObject$PrototypeDesc prototype-desc)}))
 
-(defn game-object-build-target [build-resource host-resource-node-id component-instance-datas component-build-targets]
-  {:pre [(or (nil? build-resource) (workspace/build-resource? build-resource))
+(defn game-object-build-target [source-resource host-resource-node-id component-instance-datas component-build-targets]
+  {:pre [(workspace/source-resource? source-resource)
```

Comment:
Changed to take a `source-resource` to encourage callers to obtain the `BuildResource` from the resulting `build-target`.